pkg/ans/ans_test.go

```diff
@@ -0,0 +1,236 @@
+package ans
+
+import (
+	"encoding/json"
+	"github.com/SAP/jenkins-library/pkg/xsuaa"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type Examinee struct {
+	xsuaa     *xsuaa.XSUAA
+	server    *httptest.Server
+	ans       *ANS
+	onRequest func(rw http.ResponseWriter, req *http.Request)
+}
+
+func (e *Examinee) request(rw http.ResponseWriter, req *http.Request) {
+	e.onRequest(rw, req)
+}
+
+func (e *Examinee) finish() {
+	if e.server != nil {
+		e.server.Close()
+		e.server = nil
+	}
+}
+
+func (e *Examinee) init() {
+	if e.xsuaa == nil {
+		e.xsuaa = &xsuaa.XSUAA{
+			OAuthURL:     "https://my.test.oauth.provider",
+			ClientID:     "myTestClientID",
+			ClientSecret: "super secret",
+			CachedAuthToken: xsuaa.AuthToken{
+				TokenType:   "bearer",
+				AccessToken: "1234",
+				ExpiresIn:   12345,
+			},
+		}
+	}
+	if e.server == nil {
+		e.server = httptest.NewServer(http.HandlerFunc(e.request))
```

Comment:
The httptest server is still in use, isn't it?

Reply:
Yes, for ans I did not figure out how to mock it. Only for the ans hook I could remove the tests using a server.